internal/sauceignore/matcher_test.go

```diff
@@ -0,0 +1,165 @@
+			expectedPatters: []gitignore.Pattern{},
+			expectedErr:     nil,
+		},
+		{
+			name: ".sauceignore file is exists",
+			path: file,
+			expectedPatters: []gitignore.Pattern{
+				gitignore.ParsePattern("cypress/screenshots/", nil),
+				gitignore.ParsePattern("cypress/videos/", nil),
+				gitignore.ParsePattern("node_modules/", nil),
+				gitignore.ParsePattern(".git/", nil),
+				gitignore.ParsePattern(".github/", nil),
+				gitignore.ParsePattern(".DS_Store", nil),
+			},
+			expectedErr: nil,
+		},
+	}
+
+	for _, tc := range testsCases {
+		t.Run(tc.name, func(t *testing.T) {
+			gotPatters, err := patternsFromFile(tc.path)
+			assert.Equal(t, err, tc.expectedErr)
+			assert.Equal(t, len(gotPatters), len(tc.expectedPatters))
+		})
+	}
+}
+
+func TestSauceMatcher(t *testing.T) {
+	patterns := []Pattern{
+		NewPattern("cypress/videos/"),
+		NewPattern("node_modules/"),
+		NewPattern(".git/"),
+		NewPattern(".gitignore"),
+		NewPattern("cypress/test/**"),
+		NewPattern("test.txt"),
+		NewPattern("*.log"),
+		NewPattern("!README.md"),
+		NewPattern("videos/**/app.mp4"),
+	}
+
+	testCases := []struct {
+		name      string
+		path      []string
+		isDir     bool
+		isMatched bool
+	}{
+		{
+			name:      "cypress/videos folder will be ignored",
+			path:      []string{"cypress", "videos"},
+			isDir:     true,
+			isMatched: true,
+		}, {
+			name:      "node_modules/ folder will be ignored",
+			path:      []string{"node_modules"},
+			isDir:     true,
+			isMatched: true,
+		}, {
+			name:      ".git/ folder will be ignored",
+			path:      []string{".git"},
+			isDir:     true,
+			isMatched: true,
+		}, {
+			name:      ".gitignore file will be ignored",
+			path:      []string{".gitignore"},
+			isDir:     false,
+			isMatched: true,
+		}, {
+			name:      "cypress/test/** folder will be ignored",
+			path:      []string{"cypress", "test", "test2"},
+			isDir:     true,
+			isMatched: true,
+		}, {
+			name:      "test.txt file will be ignored",
+			path:      []string{"test.txt"},
+			isDir:     false,
+			isMatched: true,
+		},
+		{
+			name:      ".sauceignore file will NOT be ignored",
+			path:      []string{".sauceignore"},
+			isDir:     false,
+			isMatched: false,
+		},
+		{
+			name:      "all log files will be ignored",
+			path:      []string{"logs", "cron.log"},
+			isDir:     false,
+			isMatched: true,
+		},
+		{
+			name:      "README.md files will NOT be ignored",
+			path:      []string{"README.md"},
+			isDir:     false,
+			isMatched: false,
+		},
+		{
+			name:      "subfolder folder will be ignored",
+			path:      []string{"videos", "subfolder", "app.mp4"},
+			isDir:     false,
+			isMatched: true,
+		},
+	}
+
+	matcher := NewMatcher(patterns)
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			assert.Equal(t, matcher.Match(tc.path, tc.isDir), tc.isMatched)
+		})
+	}
+}
+
+func crtTempSauceignoreFile() (func(), string, error) {
+	content := `cypress/screenshots/
```

Comment:
For next time, something like this may be easier to implement: https://github.com/saucelabs/saucectl/blob/master/cli/command/new/cmd_test.go#L15

Reply:
sure